app/controllers/api/v1/concerns/authenticator.rb

```diff
@@ -8,6 +8,10 @@ module Authenticator
           {controller: 'search', action: 'bulk_download'},
           {controller: 'studies', action: 'generate_manifest'}
         ]
+        URL_SAFE_TOKEN_ALLOWED_ACTIONS = [
```

Comment:
I might combine `URL_SAFE_TOKEN_ALLOWED_ACTIONS` and `TOTAT_REQUIRED_ACTIONS` together into the same variable as it feels like they're doing the same thing.  You could add in the url param as well and have a `ALLOWED_ACTIONS_BY_PARAM` with `controller`, `action`, and `param_name` attributes for each entry.

Reply:
I like keeping these separate for now since they're different authentication mechanisms.  But 100% agree there's some overlap that we should address in a systematic way